app/services/optIn/SignUpSubmissionService.scala

```diff
@@ -0,0 +1,129 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.optIn
+
+import audit.AuditingService
+import audit.models.OptInAuditModel
+import auth.MtdItUser
+import connectors.itsastatus.ITSAStatusUpdateConnector
+import connectors.itsastatus.ITSAStatusUpdateConnectorModel.{ITSAStatusUpdateResponse, ITSAStatusUpdateResponseFailure}
+import enums.JourneyType.{Opt, OptInJourney}
+import models.incomeSourceDetails.TaxYear
+import models.itsaStatus.ITSAStatus
+import models.itsaStatus.ITSAStatus._
+import models.optin.{OptInContextData, OptInSessionData}
+import play.api.Logging
+import repositories.UIJourneySessionDataRepository
+import services.DateServiceInterface
+import services.optIn.core.OptInProposition
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class SignUpSubmissionService @Inject()(
+                                         auditingService: AuditingService,
+                                         dateService: DateServiceInterface,
+                                         itsaStatusUpdateConnector: ITSAStatusUpdateConnector,
+                                         optInService: OptInService,
+                                         uiJourneySessionDataRepository: UIJourneySessionDataRepository
+                                       ) extends Logging {
+
+  def getOptInSessionData()(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[OptInSessionData]] = {
+    for {
+      uiSessionData <- uiJourneySessionDataRepository.get(hc.sessionId.get.value, Opt(OptInJourney))
+      optInSessionData: Option[OptInSessionData] = uiSessionData.flatMap(_.optInSessionData)
+    } yield {
+      optInSessionData
+    }
+  }
+
+  def makeUpdateRequest(
+                         selectedSignUpYear: Option[TaxYear],
+                         currentYearItsaStatus: Option[ITSAStatus],
+                         nextYearItsaStatus: Option[ITSAStatus]
+                       )(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[ITSAStatusUpdateResponse] = {
+
+    val currentTaxYear = dateService.getCurrentTaxYear
+    val nextTaxYear = currentTaxYear.nextYear
+
+    (selectedSignUpYear, currentYearItsaStatus, nextYearItsaStatus) match {
+      case (Some(selectedTaxYear), Some(Annual), Some(Annual)) =>
+        itsaStatusUpdateConnector.optIn(taxYear = selectedTaxYear, user.nino)
+      case (None, Some(Annual), Some(nextYearStatus)) if nextYearStatus != Annual =>
+        itsaStatusUpdateConnector.optIn(taxYear = currentTaxYear, user.nino)
+      case (None, Some(currentYearStatus), Some(Annual)) if currentYearStatus != Annual =>
+        itsaStatusUpdateConnector.optIn(taxYear = nextTaxYear, user.nino)
+      case (Some(selectedTaxYear), None, None) =>
```

Comment:
I don't really understand why there are scenarios for when they don't pick a tax year? Pretty sure the user has to select a tax year to sign up for/from no matter what

Reply:
it's because we don't populate the db with the selected tax year when it's only a single year. If there is no selected tax year it breaks. The selected tax year field is only for multi years I think. During manual testing it was breaking because of this.